libs/src/service-selection/ServiceSelection.js

```diff
@@ -209,6 +209,17 @@ class ServiceSelection {
     this.unhealthy.add(service)
   }
 
+  /** Gets unhealthy set size */
+  getUnhealthySize () {
+    return this.unhealthy.size
+  }
+
+  /** Gets services size */
+  async getServicesSize () {
+    const services = await this.getServices()
```

Comment:
Since this is async (and potentially expensive, e.g. in the case of discprov since we have to go to chain to get the info) and both select and getServicesSize call this, can we consolidate the calls so that

select = () => {
 this.services = await getServices()
 ...
}

getServicesSize => this.services.length

Reply:
or maybe it can just be removed here? i think you used the other one

Reply:
good call. i'll remoe this as i implemented a getServiceSize call in ProviderSelection